pkg/engine/task/task_toggle.go

```diff
@@ -0,0 +1,66 @@
+package task
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/kudobuilder/kudo/pkg/apis/kudo/v1beta1"
+)
+
+const (
+	toggleTaskError = "ToggleTaskError"
+)
+
+// ToggleTask will apply or delete a set of given resources to the cluster based on value of Parameter. See Run method for more details.
+type ToggleTask struct {
+	Name      string
+	Parameter string
+	Resources []string
+}
+
+func (tt ToggleTask) Run(ctx Context) (bool, error) {
+	// 1. - Get the task to run
+	task, err := tt.getTask(ctx)
+	if err != nil {
+		return false, fatalExecutionError(err, toggleTaskError, ctx.Meta)
+	}
+	// 2. - Run the returned task
+	return task.Run(ctx)
+}
+
+func (tt ToggleTask) convertToTaskSpec() v1beta1.TaskSpec {
+	return v1beta1.TaskSpec{
+		ResourceTaskSpec: v1beta1.ResourceTaskSpec{
+			Resources: tt.Resources,
+		},
+	}
+}
+
+func (tt ToggleTask) getTask(ctx Context) (Tasker, error) {
```

Comment:
That's exactly how I imagined this method to look like 👏 👏 👏  My only issue would be naming, `get` prefix is not very go-like and it's strange to have a `getTask` method within a task file. Wdyt. about `func intermediateTask` to make it clear that this is an internal and intermediate one?

Reply:
right! I was between `detectTask` or `getTask` and went for simple one as detect might give a hint of a discovery mechanism. I like `intermediate` also, let me change to that one